flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/FunctionCatalog.java

```diff
@@ -621,7 +626,7 @@ public void registerTemporarySystemFunction(
     }
 
     @SuppressWarnings("unchecked")
-    private void validateAndPrepareFunction(CatalogFunction function)
+    private CatalogFunction validateAndPrepareFunction(CatalogFunction function)
```

Comment:
Just a double check, have you checked all usages to make sure we use the returned value?

Reply:
yep, all of them now use the returned value here